### DISCUSSION TITLE
Ollama returning wrong output

### DISCUSSION THREAD

**Discussion by cedricbulteel** [ragapp/ragapp#168] [Q&A]
**Describe the bug**
When I use ollama as model provider and run a model locally, I get the following output

![Scherm­afbeelding 2024-09-05 om 16 21 33](https://github.com/user-attachments/assets/1b24c02d-b186-48d0-9434-3cb354bcaa7e)

**To Reproduce**
1. select ollama as model provider
2. run ollama model locally
3. upload knowledge
4. ask question

**Expected behavior**
The correct output

Example when using Groq:
![Scherm­afbeelding 2024-09-05 om 16 26 12](https://github.com/user-attachments/assets/e23be848-e2df-48a6-87db-2484a26ce6fc)

**Desktop (please complete the following information):**
 - OS: Ubuntu 22.04.4
 - Browser: chrome

**Comment by marcusschiesser** [comment]
which model are you using for ollama? you'll need to use one that is fine-tuned on function calling. This makes all the difference

**Comment by cedricbulteel** [comment]
I used mistral 7B but funnily enough, the problem solved itself by using llama 3.1 8B. So indeed it depends on the model
